src/ast/codegen_llvm.cpp

```diff
@@ -1614,11 +1614,55 @@ void CodegenLLVM::visit(Cast &cast)
   }
 }
 
+void CodegenLLVM::compareStructure(SizedType &our_type, llvm::Type *llvm_type)
+{
+  // Validate that what we thought the struct looks like
+  // and LLVM made of it are equal to avoid issues.
+  //
+  // As the size is used throughout the semantic phase for
+  // sizing buffers and maps we have to abort if it doesn't
+  // match.
+  // But offset is only used for printing, so we can recover
+  // from that by storing the correct offset.
+  //
+  size_t our_size = our_type.size;
+  size_t llvm_size = layout_.getTypeAllocSize(llvm_type);
+
+  if (llvm_size != our_size)
+  {
+    LOG(FATAL) << "BUG: Struct size mismatch: expected: " << our_size
+               << ", real: " << llvm_size;
+  }
+
+  auto *tuple_layout = layout_.getStructLayout(
```

Comment:
nit: `layout` is more appropriate? (since this function could be used with other than tuple)